src/modules/rc_update/rc_update.cpp

```diff
@@ -598,15 +599,28 @@ void RCUpdate::UpdateManualSwitches(const hrt_abstime &timestamp_sample)
 		// slots. And finally we add half a slot width to ensure that integer rounding
 		// will take us to the correct final index.
 		const float value = _rc.channels[_param_rc_map_fltmode.get() - 1];
-		switches.mode_slot = (((((value - slot_min) * num_slots) + slot_width_half) / (slot_max - slot_min)) +
-				      slot_width_half) + 1;
+		uint8_t switch_slot = (((((value - slot_min) * num_slots) + slot_width_half) / (slot_max - slot_min)) +
+				       slot_width_half) + 1;
 
-		if (switches.mode_slot > num_slots) {
-			switches.mode_slot = num_slots;
+		if (switch_slot > num_slots) {
+			switch_slot = num_slots;
 		}
 
-	} else if (_param_rc_map_flightmode_buttons.get() > 0) {
-		switches.mode_slot = manual_control_switches_s::MODE_SLOT_NONE;
+		// Make sure the mode switch upon consistent change sets the corresponding slot for two samples to pass the simple outlier protection
+		const bool switch_stayed_then_changed = (_last_switch_slot == _last_last_switch_slot)
```

Comment:
I would add something so that it must be consecutive valid samples to avoid surprises.

Maybe clear `_last_switch_slot` and `_last_last_switch_slot` here? https://github.com/PX4/PX4-Autopilot/blob/ef7cd7b2c07e7ac24e7ca3d39025e5489d26a91d/src/modules/rc_update/rc_update.cpp#L547-L550